### PR TITLE
FirstTry

### PR DESCRIPTION
**Pull request by AltoDomino**
Co zdupcyłem?

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -13,6 +13,11 @@ module.exports = {
             test:/\.jsx?/,
             use: ["babel-loader"],
             exclude: /node-modules/
+            },
+            {
+            test:/\.css?/,
```
Comment (FutureInventor) [inline webpack.config.js]: Why did you configure css when it's not used anywhere?

**Comment by FutureInventor** [comment]
When given login and password already exist in the database, user should get information that the data already exist.
image

**Comment by FutureInventor** [comment]
User should not be able to go to the registration when is logged in. 
image

File: index.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ import React from "react";
 import {createRoot} from 'react-dom/client'
 import { App } from "./src/App";
 
+
```
Comment (FutureInventor) [inline index.js]: Unnecessary empty line.

File: server.js [hunk 3]
```diff
@@ -1,29 +1,68 @@
...
+const cache = require("node-cache");
 
-const getFiles = async ()=>{
-    const indexHtml = await fs.readFile("./index.html")
-    const indexJs = await fs.readFile("./dist/r.bundle.js")
-    return [indexHtml,indexJs]
+const storeCache = new cache();
+
+if (!storeCache.has("registrations")) {
+    storeCache.set("registrations", []);
```
Comment (FutureInventor) [inline server.js]: Is there any specific reason why you decided to use cache instead of the plain variables? One of the main advantages of cache is the possibility to set expiration time, getting data by keys etc. What you've done here is not different from using `const registrations = []`. If you want to use cache properly instead of using array, set every user as separately. 
For example, if you have mail: test@gmail.com, and password: pass124, you should use cache like this: `storeCache.set(mail, password, <some expiration time, 5 minutes will be fine in this case>)`

Comment (FutureInventor) [reply server.js]: Later on instead of looping over all records you can easily check if email and password already exist: 
`storeCache.has(mail)` <- check if mail exist in cache
and then if mail exist:
`storeCache.get(mail) === password`  <- check if password exist in cache

Comment (FutureInventor) [reply server.js]: Do not guess. Read the documentation carefully and if you understand it, try cache approach. Otherwise, just use a simple variable `const registrations = []`

File: package.json [hunk 4]
```diff
@@ -20,5 +20,12 @@
     "react-dom": "^18.2.0",
     "webpack": "^5.91.0",
     "webpack-cli": "^5.1.4"
+  },
+  "dependencies": {
+    "css-loader": "^7.1.1",
```
Comment (FutureInventor) [inline package.json]: Do you need css-loader and style-loader in this project?

**Comment by FutureInventor** [comment]
In the HTTP standard, we have different status codes for different actions. You can learn more about it [here](https://restfulapi.net/http-status-codes/). Therefore your codes should go like this: 
- successful registration (new user is created) - code 201,
- failed registration (user already exist) - code 409,
- successful login - code 200,
- failed login - code 422.

According to those codes FE should react (not to response data as it is now).

File: server.js [hunk 5]
```diff
@@ -1,29 +1,67 @@
...
+        res.write(buffer);
+        res.end();
     }
-}).listen(8008)
\ No newline at end of file
+    else if (url.includes("youLogIn")) {
+        const LoginObjectUrl = url.split('=')[1];
+        const decodedlog = JSON.parse(decodeURIComponent(LoginObjectUrl));
+        const stringify = JSON.stringify(decodedlog);
+        let loggedIn = false;
```
Comment (FutureInventor) [inline server.js:33]: This variable should be kept outside the condition (preferably in the same scope as the registrations array). At the beginning  it should be set to `false`, when we login it should be changed to `true` and when we logout it should be set to `false`.

File: src/App.jsx [hunk 6]
```diff
@@ -1,3 +1,35 @@
...
+  Route,
+} from "react-router-dom";
+import { Succesfull } from "./youLogIn";
 
-export const App = () => <div>Empty page</div>
\ No newline at end of file
+export const App = () => {
+  return (
+    <>
+        <BrowserRouter>
```
Comment (FutureInventor) [inline src/App.jsx]: Great use of the react-router 🥳💯🥇

File: src/Login.jsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+export const Login =()=>{
+const {register, handleSubmit} = useForm()
+const navigate = useNavigate()
+
+const onSubmit = async (dataLog)=>{
+  let encodedObject = encodeURIComponent(JSON.stringify(dataLog));
+  const res = await fetch(`http://localhost:8008/youLogIn=${encodedObject}`)
+  const json = await res.json()
+  if(json){
```
Comment (FutureInventor) [inline src/Login.jsx]: Check the request's status code instead of json.

File: src/Login.jsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+      navigate("/youLogIn")
+    }
+  else {
+    alert("BŁĘDNE DANE MENDO")
+}
+}
+return(
+  <div className="login">
+  <form onSubmit={handleSubmit(onSubmit)}>
+    <label>
```
Comment (FutureInventor) [inline src/Login.jsx]: Each input should have it's own label.

Comment (FutureInventor) [reply src/Login.jsx]: Also add 'for' attribute to label and 'name' to inputs.

File: src/registration.jsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+  const navigate = useNavigate()
+  
+  const onSubmit = async (dataReg) => {
+    let encodedObject = encodeURIComponent(JSON.stringify(dataReg));
+    const res = await fetch(`http://localhost:8008/Registration=${encodedObject}`)
+    const resJson = await res.json()
+    if (typeof resJson === "object" ){
+      alert("kongratulejszyn")
+    }
+    else{alert("kajak od tyłu to wciąż kajak, a ty od tyłu jestes jebany, bo ktoś juz ci mejla zajumał")}
```
Comment (FutureInventor) [inline src/registration.jsx]: ❤️

File: src/registration.jsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+export const Registration = ()=> {
+  const { register, handleSubmit } = useForm()
+  const navigate = useNavigate()
+  
+  const onSubmit = async (dataReg) => {
+    let encodedObject = encodeURIComponent(JSON.stringify(dataReg));
+    const res = await fetch(`http://localhost:8008/Registration=${encodedObject}`)
+    const resJson = await res.json()
+    if (typeof resJson === "object" ){
```
Comment (FutureInventor) [inline src/registration.jsx]: Same as in https://github.com/KAR-HUD-SOFTWARE-DEVELOPMENT/landing-page/pull/1/files#r1591481057

File: src/registration.jsx [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+      alert("kongratulejszyn")
+    }
+    else{alert("kajak od tyłu to wciąż kajak, a ty od tyłu jestes jebany, bo ktoś juz ci mejla zajumał")}
+    
+  }
+  return (
+    <div className="registration">
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <label>
```
Comment (FutureInventor) [inline src/registration.jsx]: Same as in https://github.com/KAR-HUD-SOFTWARE-DEVELOPMENT/landing-page/pull/1/files#r1591481826

File: src/Login.jsx [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+  if(json){  
+      alert("ZALOGOWANO POPRAWNIE")
+      navigate("/youLogIn")
+    }
+  else {
+    alert("BŁĘDNE DANE MENDO")
+}
+}
+return(
+  <div className="login">
```
Comment (FutureInventor) [inline src/Login.jsx]: Why did you add className here? It's not used.

File: src/youLogIn.jsx [hunk 13]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { useNavigate } from "react-router-dom"
+
+export const Succesfull =() =>{
+    const navigateLogin = useNavigate()
+
+    return (
+    <div className="success">
```
Comment (FutureInventor) [inline src/youLogIn.jsx]: Why did you add className here? It's not used.

File: src/registration.jsx [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+    const res = await fetch(`http://localhost:8008/Registration=${encodedObject}`)
+    const resJson = await res.json()
+    if (typeof resJson === "object" ){
+      alert("kongratulejszyn")
+    }
+    else{alert("kajak od tyłu to wciąż kajak, a ty od tyłu jestes jebany, bo ktoś juz ci mejla zajumał")}
+    
+  }
+  return (
+    <div className="registration">
```
Comment (FutureInventor) [inline src/registration.jsx]: Why did you add className here? It's not used.

**Comment by FutureInventor** [comment]
After successful registration, the user should be moved automatically to the log-in form.

File: src/youLogIn.jsx [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+        <div>myślisz że jesteś fajny? chuj ci w dupe cfelu</div>
+        <button onClick={()=>navigateLogin(-1)}> WYLOGUJ
```
Comment (FutureInventor) [inline src/youLogIn.jsx]: Why '-1'?

**Comment by FutureInventor** [comment]
Reloading the page does not work after log-in.

**Review by mechupieniach — APPROVED** [review]
ok

**Comment by AltoDomino** [comment]
> Reloading the page does not work after log-in.

So what should i do to fix it? Some changes on backend?

**Comment by FutureInventor** [comment]
> > Reloading the page does not work after log-in.
> 
> So what should i do to fix it? Some changes on backend?

Tip: check what is going on on the backend server when you click refresh button in the browser. I reccomned to use debbuger.

**Comment by AltoDomino** [comment]
> > > Reloading the page does not work after log-in.
> > 
> > 
> > So what should i do to fix it? Some changes on backend?
> 
> Tip: check what is going on on the backend server when you click refresh button in the browser. I reccomned to use debbuger.

the browser suggests using useeffect instead of navigate, is this a good idea? or is the frontend okey? and the change is only required in the backend?

**Comment by FutureInventor** [comment]
> > > > Reloading the page does not work after log-in.
> > > 
> > > 
> > > So what should i do to fix it? Some changes on backend?
> > 
> > 
> > Tip: check what is going on on the backend server when you click refresh button in the browser. I reccomned to use debbuger.
> 
> the browser suggests using useeffect instead of navigate, is this a good idea? or is the frontend okey? and the change is only required in the backend?

Do not look at the FE for now. Look at backend. At which point it breaking now?

**Review by FutureInventor — APPROVED** [review]
Great job 🎉
